homeassistant/components/blueprint/importer.py

```diff
@@ -245,13 +245,32 @@ async def fetch_blueprint_from_website_url(
     return ImportedBlueprint(suggested_filename, raw_yaml, blueprint)
 
 
+async def fetch_blueprint_from_generic_url(
+    hass: HomeAssistant, url: str
+) -> ImportedBlueprint:
+    """Get a blueprint from a generic website."""
+    session = aiohttp_client.async_get_clientsession(hass)
+
+    resp = await session.get(url, raise_for_status=True)
+    raw_yaml = await resp.text()
+    data = yaml.parse_yaml(raw_yaml)
+
+    assert isinstance(data, dict)
+    blueprint = Blueprint(data)
+
+    parsed_import_url = yarl.URL(url)
+    suggested_filename = f"{parsed_import_url.host}/{parsed_import_url.parts[-1][:-5]}"
+    return ImportedBlueprint(suggested_filename, raw_yaml, blueprint)
+
+
 async def fetch_blueprint_from_url(hass: HomeAssistant, url: str) -> ImportedBlueprint:
     """Get a blueprint from a url."""
     for func in (
         fetch_blueprint_from_community_post,
         fetch_blueprint_from_github_url,
         fetch_blueprint_from_github_gist_url,
         fetch_blueprint_from_website_url,
+        fetch_blueprint_from_generic_url,
```

Comment:
It's important that the generic downloader is tried last. Can we make it less likely that the order is accidentally changed? Maybe move the functions to a global and assert in a test that `fetch_blueprint_from_generic_url` is the last element of the tuple?

Reply:
I agree, makes sense :+1: 

I've given it a shot, not a python expert, lmk if this is an appropriate way to solve this.